PushApp.ts

```diff
@@ -0,0 +1,125 @@
+import {
+    IAppAccessors,
+    ILogger,
+    IConfigurationExtend,
+    IEnvironmentRead,
+} from '@rocket.chat/apps-engine/definition/accessors';
+import { App } from '@rocket.chat/apps-engine/definition/App';
+import { IAppInfo } from '@rocket.chat/apps-engine/definition/metadata';
+import { SettingType } from '@rocket.chat/apps-engine/definition/settings';
+import { Test } from './commands/Test';
+import { ApiVisibility, ApiSecurity } from '@rocket.chat/apps-engine/definition/api';
+import { RocketEndpoint } from './endpoints/RocketEndpoint';
+import { PushEndpoint } from './endpoints/PushEndpoint';
+
+export class PushApp extends App {
+    constructor(info: IAppInfo, logger: ILogger, accessors: IAppAccessors) {
+        super(info, logger, accessors);
+
+    }
+
+    async extendConfiguration(configuration: IConfigurationExtend, environmentRead: IEnvironmentRead): Promise<void> {
+
+        await configuration.api.provideApi({
+            visibility: ApiVisibility.PUBLIC,
+            security: ApiSecurity.UNSECURE,
+            endpoints: [
+                new RocketEndpoint(this),
+                new PushEndpoint(this)
+            ]
+        })
+
+        await configuration.settings.provideSetting({
+            id:  "rc_ptkn",
```

Comment:
Bad smell: constant hardcoding string; please put that key to something like:
```
const CONFIG_RC_ACCESS_TOKEN = 'rc_ptkn';
```
Then if that key need to be changed, will be changed at a only point of code.